packages/client/src/applications/index.ts

```diff
@@ -586,14 +618,10 @@ async function updateFieldAgentDeclaredApplicationsByUser(
   const state = getState()
   const scope = getScope(state)
 
-  if (
-    !state.applicationsState.applications ||
-    state.applicationsState.applications.length !== 0 ||
-    !scope ||
-    !scope.includes('declare')
-  ) {
+  if (!scope || !scope.includes('declare')) {
```

Comment:
I don't think, we should remove the length check here. It means we will be forcing field agents to pull application from elastic everytime they log in.

Reply:
Agreed, I am going to amend the approach. Thanks @mushrafulhoque-dsi